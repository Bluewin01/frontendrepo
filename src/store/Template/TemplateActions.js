import * as types from "./TemplateTypes";

export const saveTemplateRequest = (payload) => ({
  type: types.SAVE_TEMPLATE_REQUEST,
  payload: payload,
});

export const saveTemplateRequestFailed = (error) => ({
  type: types.SAVE_TEMPLATE_REQUEST_FAILED,
  error,
});

export const saveTemplateRequestSuccess = (templateList, response) => ({
  type: types.SAVE_TEMPLATE_REQUEST_SUCCESS,
  templateList,
  response,
});

export const deleteTemplateRequest = (payload) => {
  return {
    type: types.DELETE_TEMPLATE_REQUEST,
    payload: payload,
  };
};

export const deleteTemplateRequestFailed = (error) => ({
  type: types.DELETE_TEMPLATE_REQUEST_FAILED,
  error,
});

export const deleteTemplateRequestSuccess = (templateList, response) => ({
  type: types.DELETE_TEMPLATE_REQUEST_SUCCESS,
  templateList,
  response,
});

export const getTemplateListRequest = (payload) => ({
  type: types.GET_TEMPLATE_LIST_REQUEST,
  payload: payload,
});

export const getTemplateListRequestSuccess = (templateList) => ({
  type: types.GET_TEMPLATE_LIST_REQUEST_SUCCESS,
  templateList: templateList,
});

export const getTemplateListRequestFailed = (error) => ({
  type: types.GET_TEMPLATE_LIST_REQUEST_FAILED,
  error,
});

export const setTemplateTypeRequest = (templateType) => ({
  type: types.SET_TEMPLATE_TYPE_REQUEST,
  templateType,
});

export const sendEmailRequest = (payload) => ({
  type: types.SEND_EMAIL_REQUEST,
  payload: payload,
});

export const sendEmailRequestSuccess = (response) => ({
  type: types.SEND_EMAIL_REQUEST_SUCCESS,
  response,
});

export const sendEmailRequestFailed = (error) => ({
  type: types.SEND_EMAIL_REQUEST_FAILED,
  error,
});

export const sendSmsRequest = (payload) => ({
  type: types.SEND_SMS_REQUEST,
  payload: payload,
});

export const sendSmsRequestSuccess = (response) => ({
  type: types.SEND_SMS_REQUEST_SUCCESS,
  response,
});

export const sendSmsRequestFailed = (error) => ({
  type: types.SEND_SMS_REQUEST_FAILED,
  error,
});

export const setResponseToDefault = () => ({
  type: types.SET_RESPONSE_TO_DEFAULT,
  response: null,
});

export const setErrorToDefault = () => ({
  type: types.SET_ERROR_TO_DEFAULT,
  error: null,
});

export const setTemplateObjectToEmpty = () => ({
  type: types.SET_TEMPLATE_OBJECT_TO_EMPTY,
});

export const viewTemplateRequest = (payload) => ({
  type: types.VIEW_TEMPLATE_REQUEST,
  payload,
});

export const viewTemplateRequestSuccess = (templateObject) => ({
  type: types.VIEW_TEMPLATE_REQUEST_SUCCESS,
  templateObject,
});

export const viewTemplateRequestFailed = (error) => ({
  type: types.VIEW_TEMPLATE_REQUEST_FAILED,
  error,
});

export const editTemplateRequest = (payload) => ({
  type: types.EDIT_TEMPLATE_REQUEST,
  payload,
});

export const editTemplateRequestSuccess = (payload) => ({
  type: types.EDIT_TEMPLATE_REQUEST_SUCCESS,
  templateList: payload,
});

export const editTemplateRequestFailed = (error) => ({
  type: types.EDIT_TEMPLATE_REQUEST_FAILED,
  error,
});

export const setReady = () => ({
  type: types.SET_READY,
  ready: true,
});
