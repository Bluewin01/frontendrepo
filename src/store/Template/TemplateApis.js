import axios from "axios";
import { axiosInstance } from "../../network/apis";
import AuthAPI from "../../store/Auth/AuthApis";

const handlerEnabled = false;

const templateApis = async (url, payload, refreshedToken) => {
  try {
    const apiResponse = await axiosInstance.post(url, payload, {
      handlerEnabled,
    });
    return { newToken: null, apiResponse };
  } catch (error) {
    const { message } = error.response.data;
    if (message === "jwt expired") {
      const request = error.config;
      const newToken = await AuthAPI.refreshToken({ token: refreshedToken });
      const { token } = newToken.data.data;
      request.headers.Authorization = `Bearer ${token}`;
      const apiResponse = await axios(request);
      const response = {
        newToken,
        apiResponse,
      };
      return response;
    }
    throw error;
  }
};

const getTemplateApis = async (url, payload, refreshedToken) => {
  try {
    const { type, templateId } = payload;
    let apiResponse = "";
    if (type !== undefined) {
      apiResponse = await axiosInstance.get(`${url}/${type}`, {
        handlerEnabled,
      });
    } else {
      apiResponse = await axiosInstance.get(`${url}/${templateId}`, {
        handlerEnabled,
      });
    }
    return { newToken: null, apiResponse };
  } catch (error) {
    if (error.hasOwnProperty("response") && error.response !== undefined) {
      const { message } = error.response.data;
      if (message === "jwt expired") {
        const request = error.config;
        const newToken = await AuthAPI.refreshToken({ token: refreshedToken });
        const { token } = newToken.data.data;
        request.headers.Authorization = `Bearer ${token}`;
        const apiResponse = await axios(request);
        const response = {
          newToken,
          apiResponse,
        };
        return response;
      }
      throw error;
    }
  }
};

const putTemplateApis = async (url, payload, refreshedToken) => {
  try {
    const apiResponse = await axiosInstance.put(url, payload, {
      handlerEnabled,
    });
    return { newToken: null, apiResponse };
  } catch (error) {
    const { message } = error.response.data;
    if (message === "jwt expired") {
      const request = error.config;
      const newToken = await AuthAPI.refreshToken({ token: refreshedToken });
      const { token } = newToken.data.data;
      request.headers.Authorization = `Bearer ${token}`;
      const apiResponse = await axios(request);
      const response = {
        newToken,
        apiResponse,
      };
      return response;
    }
    throw error;
  }
};

const deleteTemplateApis = async (url, payload, refreshedToken) => {
  try {
    const { id, type } = payload;
    const apiResponse = await axiosInstance.delete(`${url}/${id}/${type}`, {
      handlerEnabled,
    });
    return { newToken: null, apiResponse };
  } catch (error) {
    if (error.hasOwnProperty("response") && error.response !== undefined) {
      const { message } = error.response.data;
      if (message === "jwt expired") {
        const request = error.config;
        const newToken = await AuthAPI.refreshToken({ token: refreshedToken });
        const { token } = newToken.data.data;
        request.headers.Authorization = `Bearer ${token}`;
        const apiResponse = await axios(request);
        const response = {
          newToken,
          apiResponse,
        };
        return response;
      }
      throw error;
    }
  }
};

export default {
  templateApis,
  getTemplateApis,
  putTemplateApis,
  deleteTemplateApis,
};
