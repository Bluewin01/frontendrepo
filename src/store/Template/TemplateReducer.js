import * as types from "./TemplateTypes";

const INITIAL_STATE = {
  error: null,
  loading: false,
  templateType: "Email",
  editMode: false,
  templateObject: null,
  templateList: [],
  response: null,
  ready: false,
};

export default (state = INITIAL_STATE, action) => {
  switch (action.type) {
    case types.SAVE_TEMPLATE_REQUEST: {
      return {
        ...state,
        error: null,
        loading: true,
      };
    }
    case types.SAVE_TEMPLATE_REQUEST_FAILED: {
      return {
        ...state,
        ...action.error,
        loading: false,
      };
    }
    case types.SAVE_TEMPLATE_REQUEST_SUCCESS: {
      return {
        ...state,
        ...action.templateList,
        ...action.response,
        error: null,
      };
    }
    case types.DELETE_TEMPLATE_REQUEST: {
      return {
        ...state,
        error: null,
        loading: true,
      };
    }
    case types.DELETE_TEMPLATE_REQUEST_FAILED: {
      return {
        ...state,
        ...action.error,
        loading: false,
      };
    }
    case types.DELETE_TEMPLATE_REQUEST_SUCCESS: {
      return {
        ...state,
        ...action.templateList,
        ...action.response,
        error: null,
      };
    }
    case types.GET_TEMPLATE_LIST_REQUEST_SUCCESS: {
      return {
        ...state,
        ...action.templateList,
        error: null,
      };
    }
    case types.GET_TEMPLATE_LIST_REQUEST_FAILED: {
      return {
        ...state,
        ...action.error,
        loading: false,
      };
    }
    case types.SET_TEMPLATE_TYPE_REQUEST: {
      return {
        ...state,
        ...action.templateType,
      };
    }
    case types.SEND_EMAIL_REQUEST_SUCCESS: {
      return {
        ...state,
        ...action.response,
        error: null,
      };
    }
    case types.SEND_EMAIL_REQUEST_FAILED: {
      return {
        ...state,
        ...action.error,
        loading: false,
      };
    }
    case types.SEND_SMS_REQUEST_SUCCESS: {
      return {
        ...state,
        ...action.response,
        error: null,
      };
    }
    case types.SEND_SMS_REQUEST_FAILED: {
      return {
        ...state,
        ...action.error,
        loading: false,
      };
    }
    case types.SET_RESPONSE_TO_DEFAULT: {
      return {
        ...state,
        response: null,
      };
    }
    case types.SET_ERROR_TO_DEFAULT: {
      return {
        ...state,
        error: null,
      };
    }
    case types.SET_TEMPLATE_OBJECT_TO_EMPTY: {
      return {
        ...state,
        templateObject: null,
      };
    }
    case types.VIEW_TEMPLATE_REQUEST_SUCCESS: {
      return {
        ...state,
        ...action.templateObject,
      };
    }
    case types.VIEW_TEMPLATE_REQUEST_FAILED: {
      return {
        ...state,
        ...action.error,
        loading: false,
      };
    }
    case types.EDIT_TEMPLATE_REQUEST_SUCCESS: {
      return {
        ...state,
        ...action.templateList,
      };
    }
    case types.EDIT_TEMPLATE_REQUEST_FAILED: {
      return {
        ...state,
        ...action.error,
        loading: false,
      };
    }
    case types.SET_READY: {
      return {
        ...state,
        ready: true,
      };
    }

    default:
      return state;
  }
};
