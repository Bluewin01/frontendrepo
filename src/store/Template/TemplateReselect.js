import { createSelector } from "reselect";

const selectTemplate = (state) => state.Template;

export const makeSelectTemplateList = () =>
  createSelector(selectTemplate, (TemplateState) => TemplateState.templateList);

export const makeSelectTemplateType = () =>
  createSelector(selectTemplate, (TemplateState) => TemplateState.templateType);

export const makeSelectError = () =>
  createSelector(selectTemplate, (TemplateState) => TemplateState.error);

export const makeSelectResponse = () =>
  createSelector(selectTemplate, (TemplateState) => TemplateState.response);

export const makeSelectReady = () =>
  createSelector(selectTemplate, (TemplateState) => TemplateState.ready);

export const makeSelectTemplateObject = () =>
  createSelector(
    selectTemplate,
    (TemplateState) => TemplateState.templateObject
  );
