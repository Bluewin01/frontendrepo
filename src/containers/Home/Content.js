import React, { useState, useEffect } from "react";
import "./_HomeStyle.css";
import History from "../../routes/History";
import sampleTemplateList from "../../assets/sampleTemplate/sampleTemplate.json";
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";
import {
  makeSelectTemplateList,
  makeSelectTemplateType,
} from "../../store/Template/TemplateReselect";
import PropTypes from "prop-types";
import { Card } from "../../components/Card/Card";
import * as TemplateActions from "../../store/Template/TemplateActions";
import { Modal } from "../../components/Modal/Modal";
import "react-toastify/dist/ReactToastify.css";

const Content = (props) => {
  const {
    templateType,
    templateList,
    initiateViewTemplate,
    initiateDeleteTemplate,
  } = props;
  const sampleTemplate = sampleTemplateList.filter((sample) => {
    return sample.type === templateType;
  });
  const [showDeleteModal, setShowDeleteModal] = useState(false);
  const [selectedTemplateId, setSelectedTemplateId] = useState("");

  const createTemplate = () => {
    if (templateType === "Email") {
      History.push("/email");
    }
    if (templateType === "Sms") {
      History.push("/sms");
    }
    if (templateType === "Push") {
      History.push("/push");
    }
  };

  const editTemplate = (templateId) => {
    if (templateType === "Email") {
      initiateViewTemplate(templateId);
      History.push("/email");
    }
    if (templateType === "Sms") {
      initiateViewTemplate(templateId);
      History.push("/sms");
    }
    if (templateType === "Push") {
      initiateViewTemplate(templateId);
      History.push("/push");
    }
  };

  const deleteTemplate = () => {
    initiateDeleteTemplate(selectedTemplateId, templateType);
    setShowDeleteModal(false);
  };

  return (
    <section className="content">
      <label>Sample Templates</label>
      <div className="templateCategory">
        {sampleTemplate.map((sample) => (
          <Card
            key={sample.id}
            cardType={"templateList"}
            template={sample}
            deleteTemplate={deleteTemplate}
            buttons={[
              {
                name: "Use Template",
                onClick: () => alert("This feature is not available yet"),
              },
            ]}
          />
        ))}
      </div>
      <label>My Templates</label>
      <div className="templateCategory">
        <Card cardType={"createTemplate"} createTemplate={createTemplate} />
        {templateList.map((template) => (
          <Card
            key={template.id}
            cardType={"templateList"}
            template={template}
            buttons={[
              { name: "Edit", onClick: editTemplate },
              {
                name: "Delete",
                onClick: () => {
                  setShowDeleteModal(true);
                  setSelectedTemplateId(template.id);
                },
              },
            ]}
          />
        ))}
      </div>
      {showDeleteModal === true ? (
        <Modal
          headerTitle={`Are you sure to delete ${templateType} template?`}
          showBody={false}
          buttons={[
            {
              name: "Confirm",
              color: "danger",
              handler: deleteTemplate,
            },
            {
              name: "Cancel",
              color: "light",
              handler: () => setShowDeleteModal(false),
            },
          ]}
        />
      ) : null}
    </section>
  );
};

const mapStateToProps = createStructuredSelector({
  templateType: makeSelectTemplateType(),
  templateList: makeSelectTemplateList(),
});

const mapDispatchToProps = (dispatch) => {
  return {
    initiateViewTemplate: (templateId) =>
      dispatch(
        TemplateActions.viewTemplateRequest({
          templateId,
        })
      ),
    initiateDeleteTemplate: (templateId, templateType) =>
      dispatch(
        TemplateActions.deleteTemplateRequest({
          templateId,
          templateType,
        })
      ),
  };
};

Content.propTypes = {
  templateType: PropTypes.string,
  templateList: PropTypes.array,
  initiateViewTemplate: PropTypes.func,
  initiateDeleteTemplate: PropTypes.func,
};

export default connect(mapStateToProps, mapDispatchToProps)(Content);
