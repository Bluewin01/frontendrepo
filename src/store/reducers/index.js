import { combineReducers } from "redux";
import { configureStore } from "@reduxjs/toolkit";
import lang from "../Lang/LangReducer";
import loader from "../Loader/LoaderReducer";
import Auth from "../Auth/AuthReducer";
import Template from "../Template/TemplateReducer";

const rootReducer = combineReducers({
  lang,
  loader,
  Auth,
  Template,
});

export const setupStore = (preloadedState) => {
  return configureStore({
    reducer: rootReducer,
    preloadedState,
  });
};

export default rootReducer;
