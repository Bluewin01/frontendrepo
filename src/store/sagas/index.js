import { AuthSaga } from "../Auth/AuthSaga";
import { TemplateSaga } from "../Template/TemplateSaga";
import { fork, all } from "redux-saga/effects";

export function* watchSagas() {
  yield all([AuthSaga(), TemplateSaga()]);
}
