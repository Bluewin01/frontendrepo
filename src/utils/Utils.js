// test-utils.jsx
import React from "react";
import { render as rtlRender } from "@testing-library/react";
import { configureStore } from "@reduxjs/toolkit";
import { Provider } from "react-redux";
import { setupStore } from "../store/reducers";

export function render(
  ui,
  { preloadedState, store = setupStore(preloadedState), ...renderOptions } = {}
) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>;
  }

  // Return an object with the store and all of RTL's query functions
  return { store, ...rtlRender(ui, { wrapper: Wrapper, ...renderOptions }) };
}

export * from "@testing-library/react";
