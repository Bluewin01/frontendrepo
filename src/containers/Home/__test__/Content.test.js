import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import store from "../../../store";
import { Provider } from "react-redux";
import { render, fireEvent, screen, act } from "../../../utils/Utils";
import Content from "../Content";
import * as ACTIONS from "../../../store/Template/TemplateActions";

describe("Containers: Content", () => {
  beforeAll(() => {
    Object.defineProperty(window, "matchMedia", {
      writable: true,
      value: jest.fn().mockImplementation((query) => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(), // Deprecated
        removeListener: jest.fn(), // Deprecated
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      })),
    });
  });

  afterEach(cleanup);

  it("should be render correctly", () => {
    const mockUseState = () => jest.fn();
    const initiateViewTemplate = [
      {
        setSelectedTemplateId: mockUseState,
      },
    ];

    const tree = renderer
      .create(
        <Provider store={store}>
          <Content
            templateType="Email"
            templateList={[]}
            initiateViewTemplate={initiateViewTemplate}
          />
        </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should be redirect to /email route when click plus icon", () => {
    render(<Content />);

    const createTemplate = screen.getByTestId("createTemplate");
    fireEvent.click(createTemplate);

    expect(global.window.location.href).toContain("/email");
  });

  it("should be redirect to /sms route when click plus icon", () => {
    render(<Content />, {
      preloadedState: { Template: { templateType: "Sms", templateList: [] } },
    });

    const createTemplate = screen.getByTestId("createTemplate");
    fireEvent.click(createTemplate);

    expect(global.window.location.href).toContain("/sms");
  });

  it("should be redirect to /push route when click plus icon", () => {
    render(<Content />, {
      preloadedState: { Template: { templateType: "Push", templateList: [] } },
    });

    const createTemplate = screen.getByTestId("createTemplate");
    fireEvent.click(createTemplate);

    expect(global.window.location.href).toContain("/push");
  });

  it("should be redirect to /email route when click Edit button", () => {
    render(<Content />, {
      preloadedState: {
        Template: {
          templateType: "Email",
          templateList: [
            {
              id: "fakeId",
              name: "fakeName",
              _id: "fake_id",
            },
          ],
        },
      },
    });

    const editButton = screen.getByText("Edit");
    fireEvent.click(editButton);

    expect(global.window.location.href).toContain("/email");
  });

  it("should be redirect to /sms route when click Edit button", () => {
    render(<Content />, {
      preloadedState: {
        Template: {
          templateType: "Sms",
          templateList: [
            {
              id: "fakeId",
              name: "fakeName",
              _id: "fake_id",
            },
          ],
        },
      },
    });

    const editButton = screen.getByText("Edit");
    fireEvent.click(editButton);

    expect(global.window.location.href).toContain("/sms");
  });

  it("should be redirect to /push route when click Edit button", () => {
    render(<Content />, {
      preloadedState: {
        Template: {
          templateType: "Push",
          templateList: [
            {
              id: "fakeId",
              name: "fakeName",
              _id: "fake_id",
            },
          ],
        },
      },
    });

    const editButton = screen.getByText("Edit");
    fireEvent.click(editButton);

    expect(global.window.location.href).toContain("/push");
  });

  it("should be shown modal with text: Are you sure to delete Email template? when click delete button", () => {
    render(<Content />, {
      preloadedState: {
        Template: {
          templateType: "Email",
          templateList: [
            {
              id: "fakeId",
              name: "fakeName",
              _id: "fake_id",
            },
          ],
        },
      },
    });

    const deleteButton = screen.getByText("Delete");
    fireEvent.click(deleteButton);

    expect(
      screen.getByText("Are you sure to delete Email template?")
    ).toBeInTheDocument();
  });

  it("should be triggered deleteTemplateRequest action when click confirm for delete", () => {
    render(<Content />, {
      preloadedState: {
        Template: {
          templateType: "Email",
          templateList: [
            {
              id: "fakeId",
              name: "fakeName",
              _id: "fake_id",
            },
          ],
        },
      },
    });

    const deleteButton = screen.getByText("Delete");
    fireEvent.click(deleteButton);
    const confirmButton = screen.getByText("Confirm");
    fireEvent.click(confirmButton);

    const deleteTemplateRequest = jest.spyOn(ACTIONS, "deleteTemplateRequest");
    expect(deleteTemplateRequest).toBeDefined();
  });

  it("should be shown text: This feature is not available yet when click on Use Template button", () => {
    const alertMock = jest.spyOn(window, "alert").mockImplementation();

    render(<Content />, {
      preloadedState: {
        Template: {
          templateType: "Email",
          templateList: [
            {
              id: "fakeId",
              name: "fakeName",
              _id: "fake_id",
            },
          ],
        },
      },
    });

    const useTemplateButton = screen.getByText("Use Template");
    fireEvent.click(useTemplateButton);

    expect(alertMock).toHaveBeenCalledTimes(1);
  });

  it("should not be shown confirm button when click cancel for delete", () => {
    render(<Content />, {
      preloadedState: {
        Template: {
          templateType: "Email",
          templateList: [
            {
              id: "fakeId",
              name: "fakeName",
              _id: "fake_id",
            },
          ],
        },
      },
    });

    const deleteButton = screen.getByText("Delete");
    fireEvent.click(deleteButton);
    const cancelButton = screen.getByText("Cancel");
    fireEvent.click(cancelButton);

    const confirmButton = screen.queryByText("Confirm");
    expect(confirmButton).not.toBeInTheDocument();
  });

  it("should be able to mouseEnter on card", () => {
    render(<Content />, {
      preloadedState: {
        Template: {
          templateType: "Email",
          templateList: [
            {
              id: "fakeId",
              name: "fakeName",
              _id: "fake_id",
            },
          ],
        },
      },
    });

    const card = screen.getByText("fakeName");
    fireEvent.mouseEnter(card);
  });

  it("should be able to mouseLeave on card ", () => {
    render(<Content />, {
      preloadedState: {
        Template: {
          templateType: "Email",
          templateList: [
            {
              id: "fakeId",
              name: "fakeName",
              _id: "fake_id",
            },
          ],
        },
      },
    });

    const card = screen.getByText("fakeName");
    fireEvent.mouseLeave(card);
  });
});
