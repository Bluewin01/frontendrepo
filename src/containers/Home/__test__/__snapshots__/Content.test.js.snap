// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Containers: Content should be render correctly 1`] = `
<section
  className="content"
>
  <label>
    Sample Templates
  </label>
  <div
    className="templateCategory"
  >
    <div
      className="card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="cardContent"
        style={
          Object {
            "display": "block",
          }
        }
      >
        My Email Template
      </div>
      <div
        className="hiddenButton"
        style={
          Object {
            "display": "none",
          }
        }
      >
        <div
          className="cardButton"
          data-testid="editTemplate"
          onClick={[Function]}
          style={
            Object {
              "height": "160px",
            }
          }
        >
          Use Template
        </div>
      </div>
    </div>
  </div>
  <label>
    My Templates
  </label>
  <div
    className="templateCategory"
  >
    <div
      className="card"
      data-testid="createTemplate"
      onClick={[Function]}
    >
      <div
        className="cardContent"
      >
        <span
          aria-label="plus"
          className="anticon anticon-plus iconDesign"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="plus"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <defs>
              <style />
            </defs>
            <path
              d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"
            />
            <path
              d="M176 474h672q8 0 8 8v60q0 8-8 8H176q-8 0-8-8v-60q0-8 8-8z"
            />
          </svg>
        </span>
      </div>
    </div>
  </div>
</section>
`;
